src/components/form/Editor.tsx

```diff
@@ -0,0 +1,141 @@
+import { InputWrapper, Skeleton, type InputWrapperProps } from '@mantine/core';
+import { Editor, loader, useMonaco, type Monaco } from '@monaco-editor/react';
+
+import editorWorker from 'monaco-editor/esm/vs/editor/editor.worker?worker';
+import jsonWorker from 'monaco-editor/esm/vs/language/json/json.worker?worker';
+import {
+  useController,
+  useFormContext,
+  useFormState,
+  type FieldValues,
+  type UseControllerProps,
+} from 'react-hook-form';
+import { genControllerProps } from './util';
+import type { editor } from 'monaco-editor';
+import { useEffect } from 'react';
+
+type SetupMonacoProps = {
+  monaco: Monaco;
+  setError?: (err: string | null) => void;
+};
+
+const setupMonaco = (props: SetupMonacoProps) => {
```

Comment:
Is using monoco purely local and does it still need to load resources externally?

Reply:
most local, but still have sth from cdn. I will check it, if I have time.